src/framework/components/layout-group/layout-calculator.js

```diff
@@ -73,6 +73,7 @@ pc.extend(pc, function () {
         function maxExtentB(element, size) { return  size[b.size] * (1 - element.pivot[b.axis]); } // eslint-disable-line
 
         function calculateAll(allElements, layoutOptions) {
+            allElements = allElements.filter(shouldIncludeInLayout);
```

Comment:
Is there a chance this will be called very often like every frame, creating a new array every time? Also noticed below we are creating new pc.Vec2s for which we maybe could re-use some temporary vectors?

Reply:
It only gets called when there's a reflow to process during the current frame.

Reply:
Re the Vec2s – we could, but that's not related to this change.

Reply:
Opened a separate PR for the Vec2 thing: https://github.com/playcanvas/engine/pull/1210 :)